### PR TITLE
Change to use yaml scheduler for TW virtualization test suites

### PR DESCRIPTION
**Pull request by Julie-CAO**
- Use yaml scheduler instead of main.pm to schedule tests.
- Use DO_NOT_INSTALL_HOST rather than IPMI_DO_NOT_RESTART_HOST to skip host installation
- Support both ipxe bootup and pxe bootup to install host
- Install guest with media from http://download.opensuse.org/tumbleweed/repo/oss
- Fix some minor issues in guest autoyast control files
- Improve a little about needle match in host login

Related ticket: https://progress.opensuse.org/issues/96374

Verification run: 
[kvm PXE installation](http://10.67.129.54/tests/93) 
[kvm PXE installation without host installation & skip guest installation](http://10.67.129.54/tests/94)
[xen iPXE installation](http://10.67.129.54/tests/95)

@alice-suse @xguo @waynechen55 @nanzhg  @pdostal  Could you help review?

### CODE REVIEW

**Comment by alice-suse** [comment]
LGTM

File: products/opensuse/main.pm [hunk 1]
```diff
@@ -315,21 +315,6 @@ elsif (get_var('SECURITY_TEST')) {
     prepare_target();
     load_security_tests;
 }
-elsif (get_var('VIRT_AUTOTEST')) {
```
Comment (nanzhg) [inline products/opensuse/main.pm:318]: If this input parameter is removed, is there any impact for other jobs which was reference to this test section?

Comment (Julie-CAO) [reply products/opensuse/main.pm:318]: products/opensuse/main.pm is for openSUSE tests in O3 only, product/sle/main.pm for OSD is not touched, so our SLE tests are safe.

**Comment by Julie-CAO** [comment]
Are there any more comments? If no, can this pr be merged?

**Comment by alice-suse** [comment]
Seems no further comments, so merge it.

**Pull request merged**